File: app/src/main/java/org/mcxa/lightingcontrol/MainActivity.java
package org.mcxa.lightingcontrol;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.util.SparseArray;
import android.widget.Button;

import com.androidnetworking.AndroidNetworking;
import com.androidnetworking.error.ANError;
import com.androidnetworking.interfaces.OkHttpResponseListener;

import org.json.JSONException;
import org.json.JSONObject;

import butterknife.ButterKnife;
import butterknife.OnClick;
import okhttp3.Response;

public class MainActivity extends AppCompatActivity {
    private static SparseArray<int[]> codes = new SparseArray<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        codes.put(R.id.lchlon, new int[]{0,1});
        codes.put(R.id.lchloff, new int[]{0,0});
        codes.put(R.id.kchloff, new int[]{1,0});
        codes.put(R.id.kchlon, new int[]{1,1});
        codes.put(R.id.lvloff, new int[]{2,0});
        codes.put(R.id.lvlon, new int[]{2,1});
        codes.put(R.id.bchloff, new int[]{3,0});
        codes.put(R.id.bchlon, new int[]{3,1});
        codes.put(R.id.bloff, new int[]{4,0});
        codes.put(R.id.blon, new int[]{4,1});
    }

    @OnClick({ R.id.lchlon, R.id.lchloff, R.id.kchlon, R.id.kchloff,
            R.id.lvlon, R.id.lvloff, R.id.bchloff, R.id.bchlon,
            R.id.blon, R.id.bloff})
    public void sendCode(Button button) {
        JSONObject requestBody = new JSONObject();
        try {
            requestBody.put("light", codes.get(button.getId())[0]);
            requestBody.put("state", codes.get(button.getId())[1]);
        } catch (JSONException e) {
            Log.d("MAIN", "OH NO, there was a JSON exception. This clearly needs a try block!");
        }

        AndroidNetworking.post("http://[2606:a000:1127:4082:ba27:ebff:fe59:23c]/update")
                .addJSONObjectBody(requestBody)
                .build().getAsOkHttpResponse(new OkHttpResponseListener() {
            @Override
            public void onResponse(Response response) {
                Log.d("MAIN", "request succeeded");
            }

            @Override
            public void onError(ANError anError) {
                Log.d("MAIN", "request failed");
            }
        });
    }
}

File: app/src/main/java/org/mcxa/lightingcontrol/MainActivity.kt
package org.mcxa.lightingcontrol

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

import com.androidnetworking.AndroidNetworking
import com.androidnetworking.error.ANError
import com.androidnetworking.interfaces.JSONObjectRequestListener
import com.androidnetworking.interfaces.OkHttpResponseListener
import com.google.gson.JsonObject

import org.json.JSONException
import org.json.JSONObject

import okhttp3.Response

import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        office_lamp_on.setOnClickListener {
            callService("switch.office_lamp","switch", "turn_on")
        }
        office_lamp_off.setOnClickListener {
            callService("switch.office_lamp", "switch","turn_off")
        }

        office_lights_on.setOnClickListener {
            callService("switch.office_lights", "switch","turn_on")
        }
        office_lights_off.setOnClickListener {
            callService("switch.office_lights", "switch","turn_off")
        }

        living_room_lamp_on.setOnClickListener {
            callService("switch.living_room_lamp", "switch","turn_on")
        }
        living_room_lamp_off.setOnClickListener {
            callService("switch.living_room_lamp", "switch","turn_off")
        }

        humidifier_on.setOnClickListener {
            callService("switch.humidifier", "switch","turn_on")
        }
        humidifier_off.setOnClickListener {
            callService("switch.humidifier", "switch","turn_off")
        }

        bedroom_lamp_on.setOnClickListener {
            callService("switch.bedroom_lamp", "switch","turn_on")
        }
        bedroom_lamp_off.setOnClickListener {
            callService("switch.bedroom_lamp", "switch","turn_off")
        }

        front_door_lock_lock.setOnClickListener {
            callService("lock.kwikset_unknown_type_0003_id_0446_locked",
                    "lock","lock") {
                val lockStatus = (it?.get("attributes") as? JSONObject)?.get("lock_status")
                front_door_lock_status.text = "Lock Status: ${lockStatus}"
            }
        }
        front_door_lock_unlock.setOnClickListener {
            callService("lock.kwikset_unknown_type_0003_id_0446_locked", "lock","unlock")
        }
    }

    override fun onResume() {
        super.onResume()

        getStatus("lock.kwikset_unknown_type_0003_id_0446_locked") {
            val lockStatus = (it?.get("attributes") as? JSONObject)?.get("lock_status")
            front_door_lock_status.text = "Lock Status: ${lockStatus}"
        }
    }
}
